src/compiler/factory/nodeFactory.ts

```diff
@@ -7606,24 +7624,95 @@ function mergeEmitNode(sourceEmitNode: EmitNode, destEmitNode: EmitNode | undefi
         helpers,
         startsOnNewLine,
         snippetElement,
+        classThis,
+        assignedName,
     } = sourceEmitNode;
     if (!destEmitNode) destEmitNode = {} as EmitNode;
-    // We are using `.slice()` here in case `destEmitNode.leadingComments` is pushed to later.
-    if (leadingComments) destEmitNode.leadingComments = addRange(leadingComments.slice(), destEmitNode.leadingComments);
-    if (trailingComments) destEmitNode.trailingComments = addRange(trailingComments.slice(), destEmitNode.trailingComments);
-    if (flags) destEmitNode.flags = flags;
-    if (internalFlags) destEmitNode.internalFlags = internalFlags & ~InternalEmitFlags.Immutable;
-    if (commentRange) destEmitNode.commentRange = commentRange;
-    if (sourceMapRange) destEmitNode.sourceMapRange = sourceMapRange;
-    if (tokenSourceMapRanges) destEmitNode.tokenSourceMapRanges = mergeTokenSourceMapRanges(tokenSourceMapRanges, destEmitNode.tokenSourceMapRanges!);
-    if (constantValue !== undefined) destEmitNode.constantValue = constantValue;
+
+    // NOTE: We should have one or more lines her for each property in EmitNode, even if the line

```

Comment:
We can make this a compiler-enforced exhaustive check if we iterate over the keys of the `sourceEmitNode`, assert it to `keyof EmitNode` and switch on the key name.

Reply:
Iterating and using a switch is slower since it collapses a number of different property accesses each with independent LoadICs into a single megamorphic KeyedLoadIC. This is performance critical code, so I'd like to avoid adding more cost than is strictly necessary.